File: leetcode-solutions/bit-manipulation/solution-2401/solution.go
package solution2401

import "math"

func longestNiceSubarray(nums []int) (res int) {
    arr := []int{}
    sum := 0
    for _, v := range nums {
        for (sum & v) != 0 {
            sum -= arr[0]
            arr = arr[1:]
        }
        arr = append(arr, v)
        sum += v
        res = int(math.Max(float64(res), float64(len(arr))))
    }
    return
}

File: leetcode-solutions/bit-manipulation/solution-2401/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestNiceSubarray = function(nums) {
  let arr = [];
  let sum = 0;
  let res = 0;
  for (let i = 0; i < nums.length; i++) {
    let item = nums[i];
    while ((item & sum) !== 0) {
      sum -= arr.shift();
    }
    arr.push(nums[i]);
    sum += nums[i];
    res = Math.max(res, arr.length);
  }
  return res;
};
